sysutils/ntpsec/Portfile

```diff
@@ -33,8 +33,8 @@ python.versions     27
 # currently allow selecting its Python version, but since it's hard-coded
 # for 2.7, we can ignore this for now.
 
-depends_build-append    port:bison
-depends_lib-append      path:lib/libssl.dylib:openssl \
+depends_build-append    port:bison port:pkgconfig
+depends_lib-append      port:openssl \
```

Comment:
Why the change in the ssl dependency ? Was there an issue with the path based form, that would have allowed compatibility with other implementations than just openssl ?

Reply:
The only alternative I'm aware of is libressl, and libaes_siv (which it subsumes rather than using the port) is known to be incompatible with libressl.  It's highly unlikely that it was ever tested with anything other than OpenSSL, anyway.

Reply:
Jeremy will come around and change all of these back to the path dependency in the end anyway, as Apple cannot apparently use openssl due to licensing issues. I've seen him do that, en-masse, many times so far.

So unless you know for certain that it can never be compatible with libressl and never will be, might as well leave it. If it needs fixing, Jeremy will swoop by and fix it someday late at night.

Reply:
That makes no sense.  Why does Apple's inability to distribute OpenSSL have anything to do with whether software not distributed by Apple can depend on it?

And the "never" part also makes no sense.  If it becomes compatible with LibreSSL at some time in the future, the issue could be revisited then.  It's known not to be compatible with it **now**.  I added a comment to that effect.

The whole libaes_siv thing is somewhat interesting in its own right.  OpenSSL has included AES_SIV support for a while, but in a deficient/inefficient manner.  That's why Daniel Franke created the separate libaes_siv package, which leverages some of the OpenSSL mechanisms in a way that's not compatible with LibreSSL.  When I created the libaes_siv port (expecting it to be needed by ntpsec), I actually tried it with LibreSSL and it didn't build.  Hence:

`$ port cat libaes_siv|grep ssl`
`# NOTE: doesn't work with libressl`
`depends_lib-append  port:openssl`

That seems to have survived Jeremy so far.

I cheated and didn't retest on all platforms after adding the one comment line to the Portfile. :-)

Reply:
Ah - not worth the argument. Go ahead and change it. Just don't be surprised if it gets changed back, like the other 100,000 times this has happened.

Reply:
> That makes no sense. Why does Apple's inability to distribute OpenSSL have anything to do with whether software not distributed by Apple can depend on it?

Well -- now that jeremy is not around so much any more, perhaps it won't matter. He never explained it to me in detail -- I just saw it happen over and over. I think he is legally unable to look at the openssl code -- perhaps even to work with it -- as part of his job with Apple. 

Jeremy uses (or perhaps "used" might be a better term) many of the MacPorts libraries for his projects, sometimes personal (XQuartz) and at least in the older days, sometimes related to Apple things. I think that is why he was always so careful about changing every port in MacPorts to always be able to build with libressl, and he would go around and make changes in various ports to make them compatible with libressl if they were not.

In this particular port, I see he never changed it -- it was always a path dep, right from the beginning. But there are many other examples where it was changed to a path dep.

Now we never used to do things that would purposefully p*ss Jeremy off -- after all, what other open source project in Darwin-land had their own personal Apple engineer overseeing everything? But he's not around, so maybe it doesn't matter any more. 

I certainly don't care one way or the other as it is totally meaningless to me. 


> I cheated and didn't retest on all platforms after adding the one comment line to the Portfile. :-)

You do a lot of testing on the various systems, as do I -- that's much appreciated, rather than the "push it and watch for wreckage" approach :>